presto-verifier/src/main/java/com/facebook/presto/verifier/checksum/ArrayColumnChecksum.java

```diff
@@ -25,12 +32,43 @@ public class ArrayColumnChecksum
     private final Object checksum;
     private final Object cardinalityChecksum;
     private final long cardinalitySum;
+    // For array(floating point) we have extra aggregations collected.
+    private final FloatingPointColumnChecksum floatingPointChecksum;
 
-    public ArrayColumnChecksum(@Nullable Object checksum, @Nullable Object cardinalityChecksum, long cardinalitySum)
+    // Constructor for array(non- floating point) types.
+    public ArrayColumnChecksum(
```

Comment:
it would be better to have one constructor that does the assignment and the other constructors call that one and pass in whatever args are needed.

Reply:
I did this for FloatingPointColumnChecksum, but for ArrayColumnChecksum I can only do this for empty ctor, as we need to initialize one member as optional in some cases.

Reply:
I think providing two constructors here could be more straightforward

```
public ArrayColumnChecksum(@Nullable Object checksum, @Nullable Object cardinalityChecksum, long cardinalitySum)
{
this(checksum, cardinalityChecksum, cardinalitySum, Optional.empty());
}
```

```
public ArrayColumnChecksum(
@Nullable Object checksum,
@Nullable Object cardinalityChecksum,
long cardinalitySum,
Optional<FloatingPointColumnChecksum> floatingPointChecksum
)
{
// assignment
}
```

Reply:
@kewang1024 
> I think providing two constructors here could be more straightforward

```
Hmm, not sure if looked at the new code or the old one?
We have 3 ctors now:

    // Constructor for array(non-floating point) types.
    public ArrayColumnChecksum(
            @Nullable Object checksum,
            @Nullable Object cardinalityChecksum,
            long cardinalitySum)

    // Constructor for array(floating point) types.
    public ArrayColumnChecksum(
            @Nullable Object sum,
            long nanCount,
            long positiveInfinityCount,
            long negativeInfinityCount,
            @Nullable Object cardinalityChecksum,
            long cardinalitySum)

    // Constructor to handle cases of array(floating point) when the result table is empty (sum aggregations returns null).
    public ArrayColumnChecksum()

```

Reply:
yeah, I saw that
combined with @rschlussel's suggestion, I think changing the constructors to the above two constructors can be more starightforward?